packages/adp-tooling/src/base/change-utils.ts

```diff
@@ -0,0 +1,202 @@
+import type { Editor } from 'mem-fs-editor';
+import path, { posix } from 'path';
+
+import type {
+    AnnotationsData,
+    ChangeTypes,
+    GeneratorName,
+    BaseData,
+    InboundContent,
+    PropertyValueType
+} from '../types';
+import { FolderTypes, AnnotationFileSelectType } from '../types';
+import type { Dirent } from 'fs';
+import { existsSync, readFileSync, readdirSync } from 'fs';
+
+type InboundChangeData = { filePath: string; changeWithInboundId: InboundChange | undefined };
+type InboundChange = { content?: InboundContent };
+
+/**
+ * Writes annotation changes to the specified project path using the provided `mem-fs-editor` instance.
+ *
+ * @param {string} projectPath - The root path of the project.
+ * @param {AnnotationsData} data - The data object containing information about the annotation change.
+ * @param {object} change - The annotation data change that will be written.
+ * @param {Editor} fs - The `mem-fs-editor` instance used for file operations.
+ * @returns {void}
+ */
+export function writeAnnotationChange(projectPath: string, data: AnnotationsData, change: object, fs: Editor): void {
+    try {
+        const { answers, timestamp, annotationFileName } = data;
```

Comment:
Do you pass the answers from the Yeoman generators? If yes, please make sure that the writers are not bound to yeoman and can be used separately.

Reply:
The writers can be used outside of the yeoman environment, only that you need to provide the correct data described in the interface for that particular generator type. The method you've pointed here is a helper function for the writers and it is not exported from @sap/adp-tooling/index.ts but is used internally.

Reply:
Can we find better names for these input parameters and not call them `answers` like in Yeoman? There are no questions in the writes so probably answers does not make sense.

Reply:
@testojs These writers can be used in CLI but again in the CLI there can be prompting so the user answers some questions. I am not sure what is a better name instead of `answers`. But this is not yeoman specific.

Maybe `responses`, `userInput`, `data` (which is too generic), or `outputs`?

Reply:
Changed in: [3ed676b](https://github.com/SAP/open-ux-tools/pull/1643/commits/3ed676b560a71c69eac49e8258cf84f78c948ade)